### PR TITLE
Named aggs

### PR DESCRIPTION
**Pull request by caldwecr**
Adds support for named aggregations as well as fully qualified named aggregations.

### CODE REVIEW

**Comment by caldwecr** [comment]
Related to #192

File: lib/chewy/type/mapping.rb [hunk 1]
```diff
@@ -118,6 +118,35 @@ def field *args, &block
...
+        #         { avg: { field: 'posts.rating' } }
+        #       end
+        #     end
+        #   end
+        def agg *args, &block
+          options = args.extract_options!
+          build_root unless root_object
+
+          @_agg_defs ||= {}
+          @_agg_defs[args.first] = block
```
Comment (pyromaniac) [inline lib/chewy/type/mapping.rb]: Why don't use class_attribute here to make it work correctly in case of inheritance?

Comment (caldwecr) [inline lib/chewy/type/mapping.rb]: Great feedback. Change made. @pyromaniac

**Comment by caldwecr** [comment]
looks g2g

File: lib/chewy/type/mapping.rb [hunk 2]
```diff
@@ -118,6 +119,35 @@ def field *args, &block
...
+        #         { avg: { field: 'posts.rating' } }
+        #       end
+        #     end
+        #   end
+        def agg *args, &block
+          options = args.extract_options!
+          build_root unless root_object
+
+          self._agg_defs ||= {}
+          _agg_defs[args.first] = block
```
Comment (pyromaniac) [inline lib/chewy/type/mapping.rb]: Well, it still modifies parent hash, you have to do `self._agg_defs = _agg_defs.merge(args.first => block)`
And also move `self._agg_defs = {}` near `class_attribute :_agg_defs`. And probably you don't need `def agg_defs` anymore

**Comment by caldwecr** [comment]
@pyromaniac ok now?

**Comment by pyromaniac** [comment]
Give me some time please, I have to review this

**Comment by pyromaniac** [comment]
Ok, Let's see how would it go. I'm still not sure that named scopes can't be used instead of it, but I have at least to try to use aggregations at first

**Pull request merged**